pandas/_libs/reduction.pyx

```diff
@@ -373,7 +373,16 @@ def apply_frame_axis0(object frame, object f, object names,
                     mutated = True
             except AttributeError:
                 # `piece` might not have an index, could be e.g. an int
-                pass
+                # By definition, we are not a transform, so set mutated
+                # to True
+                mutated = True
```

Comment:
Troubleshooting some xfailed tests, I'm finding that `mutated` here is being set differently than it is in the pure-python path (inside BaseGrouper.apply).  can we pin down a canonically correct piece of code for this?

Reply:
For this PR, the relevant definition is "the result is mutated when any axis of the result is different".